src/libs/ReportUtils.js

```diff
@@ -90,6 +90,22 @@ function getChatType(report) {
     return report ? report.chatType : '';
 }
 
+/**
+ * Returns the report action with key parentReportID.
+ * This function is to be replaced by a HOC.
+ *
+ * @param {Object} parentReportActions
+ * @param {string} parentReportID
+ * @returns {Object}
+ */
+function getParentReportAction_DEV(parentReportActions, parentReportID) {
```

Comment:
This may be better moved to ReportActionsUtils.

Reply:
Sorta agree. If this is as is then yes ReportActionsUtils would be a better spot so you can use allReportActions instead of passing parentReportActions.

But i think this method and its purpose is incorrect. It says `getParentReportAction_DEV` but its getting actions for the provided reportID.

But on closer inspection it looks like the second parameter is not parentReportID and it should be `parentReportActionID`. After that change you are still getting 1 action using a specific reportActionID here, within the method there is no parent child relationship.

Okay so few things to change here, my suggestions.
1. Move this to ReportActionsUtils.
2. Update this method to accept 1param i.e., the report object (this is the child report).
3. From the method you can check if report is not a thread. If not return empty object (this should clean up some usages where you check thread and then fetch parentAction)
4. if its a thread, extract its parentReportActionID and then use allReportActions to find your specific parentReportAction. Once again if its not found it would return `{}` as the default keeping this method safe to use.
5. Also from the report object you should have both parentReportID and parentReportActionID hence making the lodash command to find specific action a lot simpler.

Reply:
also let's remove `DEV` from the method name.

Reply:
In progress of making this change for now because I agree it's way cleaner, but I thinksome calls will be replaced when we add the HOC

Reply:
After getting half way through, I realized that while I think that this is a fine change for the Util functions for the option lists (LHN/search/etc/Header), it is not a good replacement for the components that subscribe to the parent as subscribing and then passing in the parent report action list means that the component rerenders when it updates right? 
 
This point is moot when the HOC is added though so in `ReportActionItemParentAction.js` or `HeaderView`

Reply:
>  but I thinksome calls will be replaced when we add the HOC

Then the HOC that adds it should change it as necessary. Each PR is normally reviewed standalone.

> it is not a good replacement for the components..

Sorry, I'm not following the concern. So assume component ReportActionItemParentAction wants to call `getParentReportAction`. Wether its calling it in ReportUtils or in ReportActionUtils won't make a different. Plus once it receives the reportAction i don't see why it would rerender. Unless you are doing an onyx update somewhere in the middle i don't quite follow why you think the component would rerender when it calls `getParentReportAction`.

Reply:
Sorry, the problem is that we DO want it to rerender on change for the HeaderView / ReportActionItemParentAction